src/opam.nix

```diff
@@ -382,6 +384,100 @@ in rec {
       (applyOverlays overlays)
     ];
 
+  # takes an atribute set of package definitions (as produced by `queryToDefs`),
+  # deduplicates sources, and provides a list of sources to fetch
+  defsToSrcs = filterPkgs: defs:
+    let
+      nameFromURL = url: sep:
+        let
+          components = lib.strings.splitString "/" url;
+          filename = lib.last components;
+          name = builtins.head (lib.strings.splitString sep filename);
+        in name;
+      defToSrc = { version, ... }@pkgdef:
+        let
+          inherit (getUrl bootstrapPackages pkgdef) src;
+          name = let n = nameFromURL pkgdef.dev-repo "."; in
+                 # rename dune so it doesn't clash with dune file in duniverse
+                 if n == "dune" then "_dune" else n;
+        in
+        # filter out pkgs without dev-repos
+        if pkgdef ? dev-repo then { inherit name version src; } else { };
+      # remove filterPkgs
+      filteredDefs = builtins.removeAttrs defs filterPkgs;
```

Comment:
Why is this needed here? If callers of this function want to remove some packages, it's really easy for them to just call `remoteAttrs` themselves.

Reply:
My thinking was this allows packages to be removed based on their name (rather than their dev repo name). For example, the unikernel package we're has a `dummy' dev-repo which we have to remove.

Reply:
This is quite useful when we want to exclude dependencies from the monorepo that are in the same source as the unikernel (e.g. https://github.com/mirage/mirage-www) in `buildOpamMonorepo` with:
>  filterPkgs = ... (attrNames (latestVersions // query)) ...;